src/devices/pioneer_ddj_400/input.rs

```diff
@@ -1,54 +1,153 @@
 // SPDX-FileCopyrightText: The djio authors
 // SPDX-License-Identifier: MPL-2.0
 
+//! # Pioneer DDJ-400
+//!
+//! Most of the terms in this module have been taken
+//! from the manual (`DDJ-400_manual_Manual_EN.pdf`).
+//! The manual and detailed information can be found here:
+//! <https://support.pioneerdj.com/hc/en-us/sections/4416577146009-ddj-400>.
```

Comment:
https://www.pioneerdj.com/-/media/pioneerdj/software-info/controller/ddj-400/ddj-400_midi_message_list_e1.pdf

Reply:
Oh no, if i had this a few hours ago, i could have saved a lot of time :disappointed: